src/main/common/filter.c

```diff
@@ -54,82 +55,79 @@ float pt1FilterApply4(pt1Filter_t *filter, float input, uint8_t f_cut, float dT)
     return filter->state;
 }
 
-/* sets up a biquad Filter */
-void BiQuadNewLpf(float filterCutFreq, biquad_t *newState, uint32_t refreshRate)
-{
-    float sampleRate;
-
-    sampleRate = 1 / ((float)refreshRate * 0.000001f);
-
-    float omega, sn, cs, alpha;
-    float a0, a1, a2, b0, b1, b2;
-
-    /* setup variables */
-    omega = 2 * M_PIf * filterCutFreq / sampleRate;
-    sn = sinf(omega);
-    cs = cosf(omega);
-    alpha = sn * sinf(M_LN2_FLOAT /2 * BIQUAD_BANDWIDTH * omega /sn);
-
-    b0 = (1 - cs) /2;
-    b1 = 1 - cs;
-    b2 = (1 - cs) /2;
-    a0 = 1 + alpha;
-    a1 = -2 * cs;
-    a2 = 1 - alpha;
-
-    /* precompute the coefficients */
-    newState->b0 = b0 /a0;
-    newState->b1 = b1 /a0;
-    newState->b2 = b2 /a0;
-    newState->a1 = a1 /a0;
-    newState->a2 = a2 /a0;
-
-    /* zero initial samples */
-    newState->x1 = newState->x2 = 0;
-    newState->y1 = newState->y2 = 0;
-}
-
-/* Computes a biquad_t filter on a sample */
-float applyBiQuadFilter(float sample, biquad_t *state)
+static void biquadFilterInit(biquadFilter_t *filter, float filterFreq, uint32_t refreshRate, float Q, biquadFilterType_e filterType)
 {
-    float result;
+    // setup variables
+    const float sampleRate = 1 / ((float)refreshRate * 0.000001f);
+    const float omega = 2 * M_PI_FLOAT * filterFreq / sampleRate;
+    const float sn = sinf(omega);
+    const float cs = cosf(omega);
+    const float alpha = sn / (2 * Q);
+
+    float b0, b1, b2, a0, a1, a2;
+
+    switch (filterType) {
+        default:
+        case FILTER_LPF:
+            b0 = (1 - cs) / 2;
+            b1 = 1 - cs;
+            b2 = (1 - cs) / 2;
+            a0 = 1 + alpha;
+            a1 = -2 * cs;
+            a2 = 1 - alpha;
+            break;
+        case FILTER_NOTCH:
+            b0 =  1;
+            b1 = -2 * cs;
+            b2 =  1;
+            a0 =  1 + alpha;
+            a1 = -2 * cs;
+            a2 =  1 - alpha;
+            break;
+    }
 
-    /* compute result */
-    result = state->b0 * sample + state->b1 * state->x1 + state->b2 * state->x2 -
-        state->a1 * state->y1 - state->a2 * state->y2;
+    // precompute the coefficients
+    filter->b0 = b0 / a0;
+    filter->b1 = b1 / a0;
+    filter->b2 = b2 / a0;
+    filter->a1 = a1 / a0;
+    filter->a2 = a2 / a0;
 
-    /* shift x1 to x2, sample to x1 */
-    state->x2 = state->x1;
-    state->x1 = sample;
+    // zero initial samples
+    filter->d1 = filter->d2 = 0;
+}
 
-    /* shift y1 to y2, result to y1 */
-    state->y2 = state->y1;
-    state->y1 = result;
+static float biquadFilterCalculateNotchQ(uint16_t centerFreq, uint16_t cutoff) {
+    float octaves = log2f((float) centerFreq  / (float) cutoff) * 2;
+    return sqrtf(powf(2, octaves)) / (powf(2, octaves) - 1);
```

Comment:
after a bit of playing with octave:
`Q = (centerFreq  * cutoff) / (centerFreq ^ 2  - cutoff ^ 2)` is correct equation when you want to place lower 3dB at cutoff. Frequency on opposite side won't be symmetrical (center frequency is geometrical mean).
It would be better to fix the code - the difference is small, but it may confuse someone working with it ..

Another option is to specify bandwidth (`Q =  centerFreq  / bandwidth`), it may make a bit more sense (users will probably specify bandwidth as if cutoff points were symmetrical - same error in current code)